consensus/polybft/consensus_runtime.go

```diff
@@ -405,81 +423,73 @@ func (c *consensusRuntime) FSM() error {
 }
 
 // restartEpoch resets the previously run epoch and moves to the next one
-func (c *consensusRuntime) restartEpoch(header *types.Header) error {
+// returns *epochMetadata different from nil if the lastEpoch is not the current one and everything was successful
+func (c *consensusRuntime) restartEpoch(header *types.Header) (*epochMetadata, error) {
+	lastEpoch := c.epoch
+
 	systemState, err := c.getSystemState(header)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	epochNumber, err := systemState.GetEpoch()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	lastEpoch := c.getEpoch()
 	if lastEpoch != nil {
-		// Epoch might be already in memory, if its the same number do nothing.
+		// Epoch might be already in memory, if its the same number do nothing -> just return provided last one
 		// Otherwise, reset the epoch metadata and restart the async services
 		if lastEpoch.Number == epochNumber {
-			return nil
+			return lastEpoch, nil
 		}
 	}
 
 	validatorSet, err := c.config.polybftBackend.GetValidators(header.Number, nil)
 	if err != nil {
-		return err
+		return nil, err
```

Comment:
can we wrap this error with a comment to have better insights of the error?
Something similar to **"cannot get validators: %w, err"**

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1047/commits/70d5daed07acc17e48bd989c49a2921c8dc76d2a